### ISSUE TITLE
FMOD Plugin Support?

### ISSUE THREAD

**Issue by asoji** [stechyo/godot-steam-audio#65]
Howdy, I was taking a look at this project, and it's definitely an interesting project. One thing I've noticed is that the official Steam Audio has support for [FMOD as a plugin](https://github.com/ValveSoftware/steam-audio/tree/master/fmod), and I was wondering if that'll ever get added, or have some optional integration maybe with [fmod-gdextension](https://github.com/utopia-rise/fmod-gdextension) maybe?

If not, is there a way I could compile it myself? I'm not very versed in C++ but I'm willing to try a bit if needed. Thank you!

**Comment by stechyo** [comment]
Wasn't planning on adding that since I have no familiarity with FMOD. It's unlikely I'll work on that, as I would have no idea where to start.

**Comment by asoji** [comment]
Gotcha, that's fair. I generally love messing with FMOD and its banks system, and since the official source had a plugin for it, I thought it might be viable to add. If you ever do figure it out, do let me know.

**Comment by JamDoggie** [comment]
PLEASE do this. I can't live without FMOD, it makes sound design so much easier. Not being able to integrate steam audio with it means I'm going to have to roll my own solution.

**Comment by solitaryurt** [comment]
Might be better to use this GDExtension which adds FMOD support:
https://github.com/utopia-rise/fmod-gdextension

And then use the Steam Audio FMOD plugin:
https://valvesoftware.github.io/steam-audio/doc/fmod/index.html

**Comment by TristanLMartin** [comment]
Wait I'm sorry I'm confused, can you not use both the godot-steam-audio GDExtension and the fmod-gdextension and achieve what he's looking for? That was my plan but now I'm worried that I'm missing something and that won't be possible

**Comment by JamDoggie** [comment]
No. I'm not sure why I commented here since I realize it's the fmod plugin that would need plugin support. The maintainers of that plugin have said they are working on it, but it seems like life has delayed that quite a lot.

**Comment by TristanLMartin** [comment]
is there anything stopping us from using both the godot-steam-audio plugin and the fmod plugin and bridging those together in my game by grabbing those signals and things off steam-audio and passing it to FMOD? I'm really new to steam-audio and to FMOD so sorry if that doesn't make sense, trying to make sure things plug together before I dive in too deep and waste time going down the wrong path

**Comment by MonstoBusta** [comment]
The FMOD Godot plugin devs have now added FMOD Plugin support in the latest version of the fmod extension. Wwise-Godot also supports extensions now. In other words, all that remains is for Steam Audio Godot to implement support for linking to FMOD/Wwise the same way Valve's official Unity and Unreal plugins have support for linking to the FMOD and WWise plugins

**Comment by pmdevita** [comment]
It seems like it shouldn't be that complicated to get it working from here, according to [Steam Audio's docs](https://valvesoftware.github.io/steam-audio/doc/fmod/getting-started.html#integrate-steam-audio-with-your-game-engine) I think all we need to do is call `iplFMODInitialize`, `iplFMODSetHRTF`, and `iplFMODSetSimulationSettings`. Maybe we add a flag in settings to enable it?

**Comment by JamDoggie** [comment]
I released an implementation of this using C#, P/Invoke, and my own Steam Audio integration. I did have to make a small fork of the utopia rise extension, but that's just a small technical problem that can definitely be solved in a better way. I didn't want to bother making a GDExtension, so I would love to see this implemented in this project for a more seamless experience!

The repository is [here](https://github.com/JamDoggie/godot-steamaudio-dotnet/). Feel free to use whatever as it's MIT licensed.
